### ISSUE TITLE
CPU on 35-40% with H265 encoding, 100% at h264

### ISSUE THREAD

**Issue by Binnetie** [jlesage/docker-handbrake#165]
First of all, great job to make it possible to run Handbrake in a Docker container!

I'm busy to convert a few old SD series from DVD9 to H265. My new NAS (Synology DS920+) can do this job, so a can use my laptop for other things.
I installed Handbrake in a Docker en started a few converts. 
My CPU load is around 35-40%, so a encode takes around 2 hours (SD video file is 25 minutes) on slow preset in Handbrake.

H265
![h264 cpu](https://user-images.githubusercontent.com/3186149/105571592-67320e80-5d51-11eb-93bf-cda08f97b67e.JPG)

When a change the CPU priority in Docker it doesn't change anything. First i set it to medium, then to high. The CPU goes from 35-40% to 40-50% in h265.

When i choose for H264 my cpu load is 100%
![h265 cpu](https://user-images.githubusercontent.com/3186149/105571588-5da8a680-5d51-11eb-934a-90fbabd4baf3.JPG)

I would like a higher load on my CPU for H265, so it takes no 2 hours to convert a video file with a duration of 25 minutes). What can i do to fix this?

**Comment by Binnetie** [comment]
I've attached logfiles from 2 encodes with the same episode.
One in H265 with a cpu load of 40-45% en one in H264 with 100% load. 

[OPPASSEN Deel 6 DVD 1 x264 2021-01-27 11-15-10.log](https://github.com/jlesage/docker-handbrake/files/5879987/OPPASSEN.Deel.6.DVD.1.x264.2021-01-27.11-15-10.log)
[OPPASSEN Deel 6 DVD 1 x265 10bit 2021-01-27 09-06-43.log](https://github.com/jlesage/docker-handbrake/files/5879988/OPPASSEN.Deel.6.DVD.1.x265.10bit.2021-01-27.09-06-43.log)

**Comment by jlesage** [comment]
I don't think this is an issue with container.  You would probably have the same behaviour by running HandBrake on a standard Linux setup.  I've done some quick search in the past and I've find a lot of discussion about HandBrake not using all the cores.  It seems to be more evident with x265.  I've seen some comments telling that you need to encode 4K videos with x265 to get full cpu usage...

**Comment by maci2100** [comment]
Hi, Sinne ive Updated the Container, i have the same issue

**Comment by Kraveylicious** [comment]
I've got the same issue as well, .264 using 100% and .265 hovering around 30% on my Synology using the latest Handbrake image. Current version of Handbrake on macOS (native, not docker) runs both .264 and .265 at 100%.

**Comment by jlesage** [comment]
@maci2100, do you know at which image version you start seeing the behaviour change ?

I guess we should try running HandBrake "natively" on Linux to see if we have the same behaviour...

**Comment by andonovski** [comment]
Just for info, I am running latest docker image (should be 1.23.0) on a Linux host which is a VM with 3 virtual CPUs assigned.
I don't have this issue, never had it. When I encode in x265 I see all 3 virtual CPUs at 100%.

**Comment by nothing2obvi** [comment]
Just wanted to add I'm having the same issue as OP. Using latest docker image on Mac host.

**Comment by maci2100** [comment]
@jlesage the last working build is 1.17.1 

all versions above only uses less then 50% of the cpu

**Comment by Kraveylicious** [comment]
@maci2100 can confirm as well, 1.17.1 works with full CPU utilization on H.265.

**Comment by scriptchecker** [comment]
I tried using 1.17 with no success but I did raise a feature request to allow multiple encodes at the same time which would solve my use of only using 30% of CPU........

**Comment by rp1989** [comment]
I am also having the same issue on latest build 1.24.1. When using a custom x265 preset the CPU usage just doesn't go above 35%. I tried using the default preset 'H265 MKV 1080p30' and the CPU usage is maximum as set. Can you please look into this? Seems to only be an issue with CPU limit when using a custom preset.

**Comment by Binnetie** [comment]
Is there any update? Encoding from a blu-ray to X265 takes a lot of time. Are there newer versions of the container that doesn't have the problem?

**Comment by jlesage** [comment]
Can you try the latest image ?  The latest version of HandBrake seems better to max out all CPU cores when encoding :)

**Comment by NamruEU** [comment]
I have the same issue with the latest docker version 23.10.1 (d14a0665a26).

**Comment by NamruEU** [comment]
Currently running Handbrake docker container version 23.11.2 (654d24b8adc) including Handbrake 1.6.1 but the problem is still there.

**Comment by scepterus** [comment]
Same here, I never had this issue with handbrake on Windows native.

**Comment by superboo07** [comment]
Same issue as others are having here

**Comment by RandallSell** [comment]
Throwing my hat into the ring here... So I have the same issue on my QNAP TVS-h874x NAS running Container Station 3.0.9.1038 (got no idea what version of Docker it runs). However it allows me to compare a VM on the same hardware to this container. My VM is Linux Mint 21.3 Cinnamon running Handbrake 1.9. So identical virtual hardware running identical version of Handbrake. I gave 21 of the 24 cores to the Mint VM and it spanks along encoding H.265 (custom settings) and will complete in about 8 hours. FWIW at 8 cores they are all maxed at 100%. As I increase the core count the start to run close to, but not utilising 100%. With 21 cores encoding to 44 HDR H.265, they are around 70% varrying between 60% and 100%. But on average 70ish%.

The same conversion using this container takes well over 24 hours. And it is clear monitoring the resources that the container is not using all cores. But also hard to tell exactly how many it is using. CPU usage sits at around 10%. So on a 24 core system, at best it is using 2 cores (other stuff is running on the NAS).

Finally I should mention that the default setting in QNAP Container station is to allow access to all cores (unlimited). But I also tried setting it to a specific number of cores (21) to rule out what QNAP thinks (unlimited) means. But it made no difference.

So the issue is not Handbrake. The issue is then either Alpine Linux or something Container Station / Docker is doing in allocating CPU resources.

I'm not good enough with Linux to know how to check resources in a running container. But it would be good to know how many cores the container thinks it has vs how many cores Handbrake is using. In other words, is the container actually getting "unlimited" a la all 24 cores? If it is, then it is a question of why isn't Alpine allowing Handbrake to use them? If the container only has 1 or 2 cores, then clearly the issue is a Container Station / Docker issue.

2-cents
